benchmarks/triton_kernels_benchmark/flash_attention_benchmark.py

```diff
@@ -540,135 +540,180 @@ def check_close(f_val, f_ref, atol, rtol):
-        line_names=['Triton', 'XeTLA'],
-        # line styles
-        styles=[('green', '-'), ('green', '--'), ('blue', '-'), ('blue', '--')],
-        ylabel=['GB/s', 'TFlops'],  # label name for the y-axis
-        plot_name='attn-performance',
-        # name for the plot. Used also as a file name for saving the plot.
-        args={},
-    ))
-def benchmark(Z, H, N_CTX, D_HEAD, CAUSAL, MODE, provider):
-    assert MODE in ['fwd', 'bwd']
-    dtype = torch.float16
-    q = torch.randn((Z, H, N_CTX, D_HEAD), device='xpu', dtype=dtype, requires_grad=True)
-    k = torch.randn((Z, H, N_CTX, D_HEAD), device='xpu', dtype=dtype, requires_grad=True)
-    v = torch.randn((Z, H, N_CTX, D_HEAD), device='xpu', dtype=dtype, requires_grad=True)
-    sm_scale = 0.125
-    if MODE == 'bwd':
-        sm_scale = 1.3
-    quantiles = [0.5, 0.0, 1.0]
-    atol = 1e-1 if N_CTX == 16384 else 1e-2
-    # FIXME: use torch sdpa for result check after https://github.com/intel/intel-xpu-backend-for-triton/issues/2042 fixed
-    torch_fn = lambda: torch.nn.functional.scaled_dot_product_attention(q.cpu(), k.cpu(), v.cpu(
-    ), attn_mask=None, dropout_p=0.0, is_causal=CAUSAL, scale=sm_scale).to(torch.float32)
-    if MODE == 'bwd':
-        torch_o = torch_fn()
-        torch_do = torch.randn_like(torch_o)
-        torch_fn = lambda: torch_o.backward(torch_do, retain_graph=True)
-    if provider == 'onednn':
-        _, min_ms, max_ms, mean, cv = benchmark_suit.do_bench(torch_fn, n_warmup=10, n_repeat=10, quantiles=quantiles)
-
-    elif provider == 'triton':
-        triton_fn = lambda: attention(q, k, v, CAUSAL, sm_scale)
+def get_benchmark(
+    providers_filter=None,
+    fa_kernel_mode='fwd',
+    xetla_assert_result=False,
+    xetla_warn_mismatch=True,
+    **kwargs,
+):
+    supported_providers = {
+        'triton': 'Triton',
+        'xetla': 'XeTLA',
+    }
+    providers = {}
+    if providers_filter is not None:
+        for provider_key, provider_label in supported_providers.items():
+            if provider_key in providers_filter:
+                providers[provider_key] = provider_label
+    else:
+        providers = supported_providers
+
+    @benchmark_suit.perf_report(
+        benchmark_suit.Benchmark(
+            # argument names to use as an x-axis for the plot
+            x_names=['Z', 'H', 'N_CTX', 'D_HEAD', 'CAUSAL', 'MODE'],
+            x_vals=[[z, h, 16384 // z, dhead, causal, mode]
+                    for z in [1, 2, 4, 8, 16, 32]
+                    for (h, dhead) in [(16, 128), (32, 64)]
+                    for causal in [False, True]
+                    for mode in [fa_kernel_mode]]  #
+            + [[4, 48, 1024, 64, causal, mode] for causal in [False, True] for mode in [fa_kernel_mode]],
+            line_arg='provider',
+            # argument name whose value corresponds to a different line in the plot
+            # possible values for `line_arg``
+            line_vals=providers.keys(),
```

Comment:
So `Benchmark` get the expected type:
```suggestion
            line_vals=list(providers.keys()),
```

Reply:
Fixed